### PR TITLE
Add Android Build Fingerprint Client Hint

### PR DESCRIPTION
**Pull request by gbournou**
b/188060079

### CODE REVIEW

File: cobalt/browser/user_agent_platform_info.h [hunk 1]
```diff
@@ -135,7 +140,8 @@ class UserAgentPlatformInfo : public web::UserAgentPlatformInfo {
   std::string evergreen_type_;
   std::string evergreen_file_type_;
   std::string evergreen_version_;
-  std::string android_os_experience_;  // Only via Client Hints
+  std::string android_build_fingerprint_;  // Only via Client Hints
+  std::string android_os_experience_;      // Only via Client Hints
```
Comment (tphamg) [inline cobalt/browser/user_agent_platform_info.h:144]: The Cobalt extension is already generically named "os experience". Why not keep it generic in the Cobalt common code?

File: cobalt/browser/user_agent_platform_info.h [hunk 2]
```diff
@@ -135,7 +140,8 @@ class UserAgentPlatformInfo : public web::UserAgentPlatformInfo {
   std::string evergreen_type_;
   std::string evergreen_file_type_;
   std::string evergreen_version_;
-  std::string android_os_experience_;  // Only via Client Hints
+  std::string android_build_fingerprint_;  // Only via Client Hints
```
Comment (tphamg) [inline cobalt/browser/user_agent_platform_info.h:143]: Could we have a more generic name for this? Maybe something like "firmware_version_ext" or "firmware_version_details" ? (And have a similar treatment for the actual header key)

File: starboard/extension/android_info.h
Comment (tphamg) [inline starboard/extension/android_info.h:1]: I don't think we should have a platform-specific extension like this. Maybe we can make it more generic? Either include it as part of the starboard/extension/configuration.h or make a starboard/extension/platform_info.h ?

File: starboard/android/shared/android_info.cc [hunk 3]
```diff
@@ -26,6 +30,17 @@ namespace shared {
 
 namespace {
 
+bool GetBuildFingerprint(char* out_value, int value_length) {
+  JniEnvExt* env = JniEnvExt::Get();
+  ScopedLocalJavaRef<jstring> id_string(env->CallStarboardObjectMethodOrAbort(
+      "getBuildFingerprint", "()Ljava/lang/String;"));
+  std::string utf_str = env->GetStringStandardUTFOrAbort(id_string.Get());
+  if (strlen(utf_str.c_str()) + 1 > value_length)
```
Comment (zhongqiliang) [inline starboard/android/shared/android_info.cc:38]: should we truncate the value and copy value_length?

Comment (zhongqiliang) [reply starboard/android/shared/android_info.cc:38]: This is fine, if the string is longer than 1k, it should be garbage.

Comment (tphamg) [reply starboard/android/shared/android_info.cc:38]: The Android SbSystemGetProperty() function leaves the buffer unmodified when the buffer is too small -- other platforms do this as well.

File: starboard/extension/android_info.h
Comment (kaidokert) [reply starboard/extension/android_info.h:1]: Garo proposed going with a narrow scope, Android only solution here first, solving for the immediate need. Once other use cases come up on other platforms, we can add a more general version that just does a more general key/value pairs or such. IMHO this is fine as is for now - it's meant to be experimental. See [go/cobalt-client-hints-take-2](http://go/cobalt-client-hints-take-2) for background.

**Review by tphamg — APPROVED** [review]

**Review by tphamg — COMMENTED** [review]
We also have rudimentary extension tests in starboard/extension/extension_test.cc

**Review by kaidokert — APPROVED** [review]
Doesn't look wrong to me, but i'd appreciate if @zhongqiliang takes a closer look

**Review by zhongqiliang — APPROVED** [review]

**Pull request merged**